modules/dxp/apps/ai-hub/ai-hub-site-initializer/src/main/java/com/liferay/ai/hub/site/initializer/internal/workflow/kaleo/runtime/node/FixSpellingAndGrammarTaskNodeExecutorAIDelegate.java

```diff
@@ -0,0 +1,136 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.ai.hub.site.initializer.internal.workflow.kaleo.runtime.node;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.workflow.WorkflowInstanceManager;
+import com.liferay.portal.kernel.workflow.WorkflowTaskManager;
+import com.liferay.portal.workflow.kaleo.model.KaleoInstanceToken;
+import com.liferay.portal.workflow.kaleo.model.KaleoTaskInstanceToken;
+import com.liferay.portal.workflow.kaleo.runtime.ExecutionContext;
+import com.liferay.portal.workflow.kaleo.runtime.node.TaskNodeExecutorAIDelegate;
+
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.model.chat.response.ChatResponse;
+import dev.langchain4j.model.vertexai.gemini.VertexAiGeminiStreamingChatModel;
+import dev.langchain4j.service.AiServices;
+
+import java.io.Serializable;
+
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Joao Victor Alves
+ */
+@Component(service = TaskNodeExecutorAIDelegate.class)
+public class FixSpellingAndGrammarTaskNodeExecutorAIDelegate
+	implements TaskNodeExecutorAIDelegate {
+
+	@Override
+	public void execute(
+		ExecutionContext executionContext, String taskNodeName) {
+
+		Map<String, Serializable> workflowContext =
+			executionContext.getWorkflowContext();
+
+		VertexAiGeminiStreamingChatModel vertexAiGeminiStreamingChatModel =
+			VertexAiGeminiStreamingChatModel.builder(
+			).project(
+				"upgrades-accelerator-liferay"
+			).location(
+				"us-central1"
+			).modelName(
+				"gemini-2.5-flash-lite"
+			).build();
+
+		ImproveWritingTaskNodeExecutorAIDelegate.Assistant assistant =
+			AiServices.builder(
+				ImproveWritingTaskNodeExecutorAIDelegate.Assistant.class
+			).systemMessageProvider(
+				object -> StringBundler.concat(
+					"You are an expert linguistic editor. Your sole task is ",
+					"to correct all grammatical, spelling, and punctuation ",
+					"errors in the provided text while preserving its ",
+					"meaning, tone, and style. Do not alter structure or ",
+					"wording beyond what is necessary for grammatical ",
+					"precision and natural fluency. Output only the corrected ",
+					"text, with no explanations or commentary.")
```

Comment:
@MrJohn1911 what happens is there is nothing to be fixed, will it retrieve the same text? please ensure that it will retrieve the same text.